src/patch.jl

```diff
@@ -126,19 +126,25 @@ end
 ```
 """
 function apply(body::Function, pe::PatchEnv)
-    prev_pe = get_active_env()
-    set_active_env(merge(prev_pe, pe))
-    try
-        return body()
-    finally
-        set_active_env(prev_pe)
-    end
+    merged_pe = merge(PATCH_ENV[], pe)
+    return with_active_env(body, merged_pe)
 end
 
 function apply(body::Function, patches; debug::Bool=false)
     return apply(body, PatchEnv(patches, debug))
 end
 
-const PATCH_ENV = Ref{PatchEnv}(PatchEnv())
-set_active_env(pe::PatchEnv) = (PATCH_ENV[] = pe)
-get_active_env() = PATCH_ENV[]
+if isdefined(Base, :ScopedValue) # added in v"1.11"
+    const PATCH_ENV = ScopedValue(PatchEnv())
+    with_active_env(body::Function, pe::PatchEnv) = with(body, PATCH_ENV => pe)
```

Comment:
`ScopedValue` was unexported as of: https://github.com/JuliaLang/julia/pull/53004
```suggestion
    const PATCH_ENV = Base.ScopedValue(PatchEnv())
    with_active_env(body::Function, pe::PatchEnv) = Base.with(body, PATCH_ENV => pe)
```